packages/contracts-bedrock/src/interop/CrossL2Inbox.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ISemver } from "src/universal/ISemver.sol";
+
+/// @notice Entry to post to the inbox.
+///         The postie may deliver multiple entries per mail delivery.
+/// @custom:field chain   Chain identifier.
+/// @custom:field output  Output-root of the chain.
+struct InboxEntry {
+    bytes32 chain;
+    bytes32 output;
+}
+
+/// @custom:proxied
+/// @title CrossL2Inbox
+/// @notice The CrossL2Inbox receives output-roots of any chain,
+///         and makes the output-roots available for cross-L2 proving.
+contract CrossL2Inbox is ISemver {
+    /// @notice The address that is allowed to post into the inbox.
+    /// This is temporary for Interop Milestone 0:
+    /// this will be changed to a system-only address later.
+    address internal immutable SUPERCHAIN_POSTIE;
+
+    /// @notice The collection of output roots, by chain.
+    /// chain ID => output root => bool.
+    mapping(bytes32 => mapping(bytes32 => bool)) public roots;
```

Comment:
Why is chainid a `bytes32` instead of a `uint`?

Reply:
From a UX perspective it'll be easier to represent as an integer instead of a hex value

Reply:
We're thinking of versioning and formatting the chain identifiers in a way that's more forward compatible if we ever have to deal with new source/destination types, or conflicting chains.